Kernel/Prekernel/Prekernel.h

```diff
@@ -6,4 +6,32 @@
 
 #pragma once
 
+#ifdef __cplusplus
+#    include <Kernel/Multiboot.h>
+#    include <Kernel/PhysicalAddress.h>
+#    include <Kernel/VirtualAddress.h>
+#endif
+
 #define MAX_KERNEL_SIZE 0x3000000
+
+#ifdef __cplusplus
+namespace Kernel {
+
+struct [[gnu::packed]] BootInfo {
+    u32 start_of_prekernel_image;
+    u32 end_of_prekernel_image;
+    u64 kernel_base;
+    u64 multiboot_info_ptr;
+#    if ARCH(X86_64)
```

Comment:
Particular reason to move these in the middle instead of at the end?

Reply:
Nah, not really - that's just pretty much the order I added them in originally. This PR doesn't move them though.